### PR TITLE
test(spanner): verify that NUMERIC can be used as a table key

### PR DESCRIPTION
**Pull request by devbww**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6950)

### CODE REVIEW

File: google/cloud/spanner/integration_tests/data_types_integration_test.cc [hunk 1]
```diff
@@ -343,6 +343,29 @@ TEST_F(DataTypeIntegrationTest, WriteReadArrayNumeric) {
   EXPECT_THAT(*result, UnorderedElementsAreArray(data));
 }
 
+TEST_F(DataTypeIntegrationTest, InsertAndQueryWithNumericKey) {
+  // TODO(#5024): Remove this check when the emulator supports NUMERIC.
+  if (google::cloud::internal::GetEnv("SPANNER_EMULATOR_HOST").has_value()) {
```
Comment (coryan) [inline google/cloud/spanner/integration_tests/data_types_integration_test.cc:348]: For another PR maybe: the base class for these tests could add a `bool UsingEmulator() const` member function.

File: google/cloud/spanner/integration_tests/data_types_integration_test.cc [hunk 2]
```diff
@@ -343,6 +343,29 @@ TEST_F(DataTypeIntegrationTest, WriteReadArrayNumeric) {
...
+  auto commit_result = client.Commit(
+      Mutations{InsertOrUpdateMutationBuilder("NumericKey", {"Key"})
+                    .EmplaceRow(key)
+                    .Build()});
+  ASSERT_STATUS_OK(commit_result);
+
+  auto rows = client.Read("NumericKey", KeySet::All(), {"Key"});
+  using RowType = std::tuple<Numeric>;
+  auto row = GetSingularRow(StreamOf<RowType>(rows));
+  EXPECT_STATUS_OK(row);
```
Comment (coryan) [inline google/cloud/spanner/integration_tests/data_types_integration_test.cc:363]: Consider: `ASSERT_STATUS_OK(row);` instead

**Review by coryan — APPROVED** [review]

File: google/cloud/spanner/integration_tests/data_types_integration_test.cc (hunk 1 above)
Comment (devbww) [reply google/cloud/spanner/integration_tests/data_types_integration_test.cc]: Done.

File: google/cloud/spanner/integration_tests/data_types_integration_test.cc (hunk 2 above)
Comment (devbww) [reply google/cloud/spanner/integration_tests/data_types_integration_test.cc]: Done.

**Review by coryan — APPROVED** [review]
LGTM

**Pull request merged**